docs/using-avail/getting-started/light-client-embedding.md

```diff
@@ -0,0 +1,288 @@
+### Fetching decoded application data for given block
+
+After data is verified, it can be fetched with `GET` request on `/v1/appdata/{block_number}` endpoint, by specifying `decode=true` query parameter.
+
+```rust
+
+use base64::Engine as _;
+use reqwest::StatusCode;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ExtrinsicsData {
+    pub block: u32,
+    pub extrinsics: Vec<String>,
+}
+
+const LIGHT_CLIENT_URL: &str = "http://127.0.0.1:7000";
+
+let block_number = 2;
+let confidence_url = format!("{LIGHT_CLIENT_URL}/v1/appdata/{block_number}?decode=true");
+let response = reqwest::get(confidence_url).await.unwrap();
+
+if response.status() == StatusCode::OK {
+    let data: ExtrinsicsData =
+        serde_json::from_str(&response.text().await.unwrap()).unwrap();
+
+    let decoded_bytes = engine::general_purpose::STANDARD
+        .decode(&data.extrinsics[0])
+        .unwrap();
+    let decoded_string = String::from_utf8(decoded_bytes).unwrap();
+    println!("{decoded_string:?}");
+}
+
+// ...error handling...
+```
+
+
+### Waiting for application data to be verified
+
+If light client is still processing specific block, we can poll light client with `GET` request on `/v1/appdata/{block_number}` endpoint, and wait for data to became available.
+
+```rust
+use avail_subxt::primitives::AppUncheckedExtrinsic;
+use serde::{Deserialize, Serialize};
+use anyhow::anyhow;
+use reqwest::StatusCode;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ExtrinsicsData {
+    pub block: u32,
+    pub extrinsics: Vec<AppUncheckedExtrinsic>,
+}
+
+const POLLING_TIMEOUT: Duration = Duration::from_secs(120);
+const POLLING_INTERVAL: Duration = Duration::from_secs(1);
+const LIGHT_CLIENT_URL: &str = "http://127.0.0.1:7000";
+
+async fn wait_for_appdata(appdata_url: &str, block: u32) -> anyhow::Result<ExtrinsicsData> {
+    let appdata_url = format!("{LIGHT_CLIENT_URL}/v1/appdata/{block}");
+    
+    let start_time = std::time::Instant::now();
+
+    loop {
+        if start_time.elapsed() >= POLLING_TIMEOUT {
+            return Err(anyhow!("Poll timeout exceeded"));
+        }
+
+        let response = reqwest::get(&appdata_url).await?;
+
+        match response.status() {
+            // If status is OK. response continas json representation of submitted data
+            StatusCode::OK => {
+                let text = &response.text().await?;
+                return Ok(serde_json::from_str(text)?);
+            }
+            // If it is not found, there is no data for given block and application
+            StatusCode::NOT_FOUND => {
+                return Ok(ExtrinsicsData {
+                    block,
+                    extrinsics: vec![],
+                })
+            }
+            // Wait for data to became available otherwise
+            _ => {
+                tokio::time::sleep(POLLING_INTERVAL).await;
+                continue;
+            }
+        }
+    }
+}
+
+```
+
```

Comment:
Can we add some simple `curl` commands? Most examples might look cumbersome but they are simpler as `curl`/`wget` requests.

Reply:
Sure we can have both sections, rust examples and curl examples. Ill create another PR for this.